docs/running-on-kubernetes.md

```diff
@@ -307,7 +307,18 @@ And, the claim name of a `persistentVolumeClaim` with volume name `checkpointpvc
 spark.kubernetes.driver.volumes.persistentVolumeClaim.checkpointpvc.options.claimName=check-point-pvc-claim
 ```
 
-The configuration properties for mounting volumes into the executor pods use prefix `spark.kubernetes.executor.` instead of `spark.kubernetes.driver.`. For a complete list of available options for each supported type of volumes, please refer to the [Spark Properties](#spark-properties) section below.
+The configuration properties for mounting volumes into the executor pods use prefix `spark.kubernetes.executor.` instead of `spark.kubernetes.driver.`.
+
+For example, you can mount a dynamically-created persistent volume claim per executor by using `OnDemand` as a claim name and `storageClass` and `sizeLimit` options like the following. This is useful in case of [Dynamic Allocation](configuration.html#dynamic-allocation).
```

Comment:
Can you explain why it's useful for DA? Currently, this doesn't support DA yet. I think the use-case will be for people to use larger shuffle disks in Kube where larger local disks are not allowed.

Reply:
@dbtsai . What do you mean by the following?
>  Currently, this doesn't support DA yet. 

Since Apache Spark 3.0.0, dynamic allocation with K8s has been supported with shuffle data tracking. And, this feature is also developed for both additional large disk requirement and dynamic allocation scenario. For example, in case of dynamic allocation, the executor id increases monotonically and indefinitely, so users cannot prepare pre-populated PVCs.

Reply:
In short, I made the PVC(and PV) be created at every pod creation in `ExecutorPodsAllocator.scala`. Please see the original code. For example, if you kill some pod, new executor is going to be created with a new PVC.
- https://github.com/apache/spark/pull/29846 [SPARK-32971][K8S] Support dynamic PVC creation/deletion for K8s executors

cc @viirya

Reply:
Get you. Thanks!